spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/SparkV2Filters.java

```diff
@@ -289,11 +311,28 @@ private static Object convertLiteral(Literal<?> literal) {
     return literal.value();
   }
 
-  private static Expression handleEqual(String attribute, Object value) {
-    if (NaNUtil.isNaN(value)) {
-      return isNaN(attribute);
+  private static UnboundPredicate<Object> handleEqual(Pair<UnboundTerm<Object>, Object> children) {
+    if (children.second() == null) {
+      return isNull(children.first());
+    }
+
+    if (NaNUtil.isNaN(children.second())) {
+      return isNaN(children.first());
+    } else {
+      return equal(children.first(), children.second());
+    }
+  }
+
+  private static UnboundPredicate<Object> handleNotEqual(
+      Pair<UnboundTerm<Object>, Object> children) {
+    if (children.second() == null) {
+      return notNull(children.first());
+    }
+
+    if (NaNUtil.isNaN(children.second())) {
+      return notNaN(children.first());
     } else {
-      return equal(attribute, value);
+      return notEqual(children.first(), children.second());
```

Comment:
I had a question , was this originally a bug?  it was returning equal(string, value), not equal(NamedReference, value)

Reply:
(My question is about the original code)

Reply:
The string will be wrapped into `NamedReference`. There are two API as follows:
```java
  public static <T> UnboundPredicate<T> equal(String name, T value) {
    return new UnboundPredicate<>(Expression.Operation.EQ, ref(name), value);
  }

  public static <T> UnboundPredicate<T> equal(UnboundTerm<T> expr, T value) {
    return new UnboundPredicate<>(Expression.Operation.EQ, expr, value);
  }
```

Reply:
I see, and you plan to pass UnboundTerm instead of string, for following pr